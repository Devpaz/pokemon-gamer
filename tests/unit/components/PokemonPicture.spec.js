import {shallowMount} from '@vue/test-utils'
import PokemonPictures from '@/components/PokemonPicture';

describe('PokemonPicture component', () => {
    test('debe de hacer match con el snapshot', () => {

        const wrapper = shallowMount( PokemonPictures, {
            props: {
                pokemonId : 1,
                showPokemon: false
            }
        })
        expect( wrapper.html() ).toMatchSnapshot()

    })

    test('debe de mostrar la imagen oculta y el pokemon 100', () => {
        const wrapper = shallowMount( PokemonPictures, {
            props: {
                pokemonId: 100,
                showPokemon: false
            }
        })

        const [img1, img2] = wrapper.findAll('img')

        expect( img1.exists() ).toBeTruthy()
        expect( img2 ).toBe(undefined)
        expect( img1.classes('hidden-pokemon') ).toBe(true)
        expect( img1.attributes('src') ).toBe('https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/100.svg')
        

    })

    test('debe de mostrar el pokemon si showPokemon:true', () => {
        const wrapper = shallowMount( PokemonPictures, {
            props: {
                pokemonId: 100,
                showPokemon: true
            }
        })
        
        
        const img1 = wrapper.find('img')

        expect( img1.exists() ).toBeTruthy()
        expect( img1.classes('hidden-pokemon') ).toBe(false)
        expect( img1.classes('fade-in') ).toBe(true)

    })
})