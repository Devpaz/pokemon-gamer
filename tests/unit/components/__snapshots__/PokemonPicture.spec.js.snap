// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonPicture component debe de hacer match con el snapshot 1`] = `<div class="pokemon-container"><img src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/1.svg" class="hidden-pokemon" alt="pokemon"></div>`;
